module/motion/walk/TorsoMovement/src/TorsoMovement.cpp

```diff
@@ -0,0 +1,122 @@
+#include "TorsoMovement.h"
+
+#include <Eigen/Geometry>
+#include <vector>
+
+#include "extension/Configuration.h"
+
+#include "message/behaviour/ServoCommand.h"
+#include "message/input/Sensors.h"
+#include "message/motion/KinematicsModel.h"
+#include "message/motion/TorsoTarget.h"
+
+#include "utility/behaviour/Action.h"
+#include "utility/input/LimbID.h"
+#include "utility/input/ServoID.h"
+#include "utility/math/matrix/Transform3D.h"
+#include "utility/motion/InverseKinematics.h"
+
+namespace module {
+namespace motion {
+    namespace walk {
+
+        using extension::Configuration;
+
+        using message::behaviour::ServoCommand;
+        using message::input::Sensors;
+        using message::motion::KinematicsModel;
+        using message::motion::TorsoTarget;
+        using utility::behaviour::RegisterAction;
+        using utility::input::LimbID;
+        using utility::input::ServoID;
+        using utility::math::matrix::Transform3D;
+        using utility::motion::kinematics::calculateLegJoints;
+
+
+        TorsoMovement::TorsoMovement(std::unique_ptr<NUClear::Environment> environment)
+            : Reactor(std::move(environment)), subsumptionId(size_t(this) * size_t(this) - size_t(this)) {
+
+            // Left for possible use in future
+            on<Configuration>("TorsoMovement.yaml").then([this](const Configuration& config) {
+
+            });
+
+
+            on<Trigger<Sensors>, With<KinematicsModel>, With<TorsoTarget>>().then(
+                [this](const Sensors& sensors, const KinematicsModel& model, const TorsoTarget& target) {
+                    // Get support foot coordinate system
+
+                    Eigen::Affine3d Htf;  // support foot
+
+                    // Right foot is the support foot
+                    if (target.isRightFootSupport) {
+                        // Transform of right foot to torso
+                        Htf = Eigen::Affine3d(sensors.forwardKinematics[ServoID::R_ANKLE_ROLL]);
+                    }
+                    // Left foot is the support foot
+                    else {
+                        // Transform of left foot to torso
+                        Htf = Eigen::Affine3d(sensors.forwardKinematics[ServoID::L_ANKLE_ROLL]);
+                    }
+
+                    // Target given in support foot space
+                    Eigen::Affine3d Haf;
+                    Haf = target.Haf;
+
+                    // Create vector to target from support foot
+                    Eigen::Vector3d rAFf = -Haf.translation();
+                    // Position of target in torso space
+                    Eigen::Vector3d rATt = Htf * rAFf;
+                    // Create next torso target in torso space
+                    Eigen::Vector3d rT_tTt = rATt * 0.01;
```

Comment:
Should `0.01` be a configurable parameter?

Reply:
I think that's the speed of torso movement, which should be based on the difference between now and when the torso is supposed to reach it's position